controllers/grid/settings/library/form/EditLibraryFileForm.inc.php

```diff
@@ -57,6 +67,25 @@ public function initData()
      */
     public function execute(...$functionArgs)
     {
+        $userId = Application::get()->getRequest()->getUser()->getId();
+
+        // Fetch the temporary file storing the uploaded library file
+        $temporaryFileDao = DAORegistry::getDAO('TemporaryFileDAO'); /* @var $temporaryFileDao TemporaryFileDAO */
+        $temporaryFile = $temporaryFileDao->getTemporaryFile(
+            $this->getData('temporaryFileId'),
+            $userId
+        );
+        if ($temporaryFile) {
+            $libraryFileDao = DAORegistry::getDAO('LibraryFileDAO'); /* @var $libraryFileDao LibraryFileDAO */
+            $libraryFileManager = new LibraryFileManager($this->contextId);
+
+            // Convert the temporary file to a library file and store
+            $this->libraryFile = $libraryFileManager->replaceFromTemporaryFile($temporaryFile, $this->getData('fileType'), $this->libraryFile);
+            // Clean up the temporary file
+            import('lib.pkp.classes.file.TemporaryFileManager');
```

Comment:
With 3.4.0 and newer, you can...
- Add to the top: `use PKP\file\TemporaryFileManager;`
- Get rid of the `import` statement

Reply:
Done.